libs/wingc/src/lib.rs

```diff
@@ -78,11 +78,16 @@ pub unsafe extern "C" fn wingc_compile(ptr: u32, len: u32) {
 	if let Err(mut err) = results {
 		// Sort error messages by line number (ascending)
 		err.sort_by(|a, b| a.cmp(&b));
-		eprintln!(
-			"Compilation failed with {} errors\n{}",
+		let result = format!(
+			"Compilation failed with {} error(s)\n{}",
 			err.len(),
 			err.iter().map(|d| format!("{}", d)).collect::<Vec<_>>().join("\n")
 		);
+
+		let leaked = result.into_bytes().leak();
+		combine_ptr_and_length(leaked.as_ptr() as u32, leaked.len() as u32)
```

Comment:
If we have a `ptr_to_string` would it make sense to abstract this into a `string_to_ptr` function? I think there might be other places we do this as well

Reply:
Makes sense, it's used in two places and will be used in more over time